src/main/java/uwuke/input/Parser.java

```diff
@@ -0,0 +1,149 @@
+ * by this class when a string in the wrong format is passed in to any of the methods.
+ */
+public abstract class Parser {
+    /**
+     * Processes string into useful information when command is identified as a deadline command
+     * <p> Regular Expression: deadline\\s.+/by\\s.+$
+     * 
+     * @param input deadline command that matches the above regular expression
+     * @return string array of length 2, can be used directly to create a new deadline object
+     * @throws DukeException if input cannot be parsed
+     */
+    public static String[] parseDeadline(String input) throws DukeException {
+        try {
+            String removeDeadlineKeyword = input.substring(9);
+            int by = removeDeadlineKeyword.indexOf("/by");
+            String task = removeDeadlineKeyword.substring(0, by).strip();
+            String deadline = removeDeadlineKeyword.substring(by+3, removeDeadlineKeyword.length()).strip();
+            return new String[] {task, deadline};
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Unexpected input format, could not parse deadline properly.");
+        } catch (Exception e) {
+            throw new DukeException("Unexpected error occured");
+        }
+    }
+
+    /**
+     * Processes string into useful information when command is identified as a event command
+     * <p> Regular Expression: ^event\\s.+/from\\s.+/to\\s.+$
+     * 
+     * @param input event command that matches the above regular expression
+     * @return string array of length 3, can be used directly to create a new deadline object
+     * @throws DukeException if input cannot be parsed
+     */
+    public static String[] parseEvent(String input) throws DukeException {
+        try {
+            String removeEventKeyword = input.substring(6);
+            int from = removeEventKeyword.indexOf("/from");
+            int to = removeEventKeyword.indexOf("/to");
+            String task = removeEventKeyword.substring(0, from).strip();
+            String start = removeEventKeyword.substring(from+5, to).strip();
+            String end = removeEventKeyword.substring(to+3, removeEventKeyword.length()).strip();
+            return new String[] {task, start, end};
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Unexpected input format, could not parse event properly.");
+        } catch (Exception e) {
+            throw new DukeException("Unexpected error occured");
+        }
+    }
+
+    /**
+     * Processes string into useful information when command is identified as a todo command
+     * <p> Regular Expression: ^todo\\s.+$
+     * 
+     * @param input todo command that matches the above regular expression
+     * @return string that can be used directly to create a new deadline object
+     * @throws DukeException if input cannot be parsed
+     */
+    public static String parseTodo(String input) throws DukeException {
```

Comment:
Very well-written function!! But wanted to ask if you would prefer to keep `Todo` or change it to `ToDo`?

Reply:
Yes it should be `ToDo`. Good catch!